db/change_cache.go

```diff
@@ -208,13 +213,30 @@ func (c *changeCache) Start(initialSequence uint64) error {
 	return nil
 }
 
+// waitForBGTCompletion waits for all the background tasks to finish.
+func waitForBGTCompletion(ch ...chan struct{}) {
+	for _, v := range ch {
+	waitForCompletion:
+		for {
+			select {
+			case <-v:
+				break waitForCompletion
```

Comment:
Do we need to consider a timeout in the event that v never closes to prevent deadlocking?

Reply:
Yeah, that is a good idea. Introduced a timeout of 30 seconds to wait for all background tasks to terminate before completely stopping the server.